File: v1/Demo/source/demo_get_errorcode.cpp
#include "../include/Handapi.h"

using namespace HandProtocol;

int main()
{
    Handapi api;
    api.init(LEFT_IP);
    vector<long> errorcode = api.get_errorcode();

    if (errorcode.size() != 6)
    {
        Logger::get_instance()->print_trace_error("get errorcode failed\n");
        return ResultCode::FAIL;
    }
    
    std::cout<<"get errorcode success : ";
    for (int i = 0; i < errorcode.size(); ++i) {
        std::cout << errorcode[i] << " ";
    }
    std::cout << std::endl;


    return 0;
}

File: v1/Demo/demo_get_errorcode.py
import sys, os
sys.path.append(os.path.join(os.path.dirname(__file__), "../"))
from Hand_Sdk.HandSocket import *
from Hand_Sdk.Handapi import *

def main():
    ip_left = "192.168.137.39" # lefthand
    # ip_right = "192.168.137.19" # righthand
    api = API(ip_left)
    errorcode = api.get_errorcode()

    if len(errorcode) != 6:
        logger.print_trace_error("get errorcode failed")
    else:
        logger.print_trace("get errorcode success", errorcode)

if __name__ == "__main__":
    main()



